packages/neos-ui-guest-frame/src/initializeGuestFrame.js

```diff
@@ -60,13 +61,45 @@ export default ({globalRegistry, store}) => function * initializeGuestFrame() {
         return;
     }
 
-    const nodes = Object.assign({}, guestFrameWindow['@Neos.Neos.Ui:Nodes'], {
-        [documentInformation.metaData.documentNode]: documentInformation.metaData.documentNodeSerialization
+    // Load legacy node data scripts from guest frame - remove with Neos 9.0
+    const legacyNodeData = guestFrameWindow['@Neos.Neos.Ui:Nodes'] || {};
+
+    // Load all nodedata for nodes in the guest frame and filter duplicates
+    const {q} = yield backend.get();
+    const nodeContextPathsInGuestFrame = findAllNodesInGuestFrame().map(node => node.getAttribute('data-__neos-node-contextpath'));
+
+    // Filter nodes that are already present in the redux store and duplicates
+    const nodesByContextPath = store.getState().cr.nodes.byContextPath;
+    const nodesAlreadyPresentInStore = {};
+    const notFullyLoadedNodeContextPaths = [...new Set(nodeContextPathsInGuestFrame)].filter((contextPath) => {
+        const node = nodesByContextPath[contextPath];
+        const nodeIsLoaded = node !== undefined && node.isFullyLoaded;
+        if (nodeIsLoaded) {
+            nodesAlreadyPresentInStore[contextPath] = node;
+            return false;
+        }
+        return true;
     });
 
+    // Load remaining list of not fully loaded nodes from the backend if there are any
+    const fullyLoadedNodesFromContent = notFullyLoadedNodeContextPaths.length > 0 ? (yield q(notFullyLoadedNodeContextPaths).get()).reduce((nodes, node) => {
+        nodes[node.contextPath] = node;
+        return nodes;
+    }, {}) : [];
+
+    const nodes = Object.assign(
+        {},
+        legacyNodeData, // Merge legacy node data from the guest frame - remove with Neos 9.0
+        nodesAlreadyPresentInStore,
```

Comment:
nodesAlreadyPresentInStore are already present in the store so we dont need to  `actions.CR.Nodes.merge(nodes)` them? Its not a replace...

Reply:
Correct, I still need to merge them into one list for initialisation, but not merge them into the store. Thx!